examples/deserialize-transaction/src/example.ts

```diff
@@ -0,0 +1,403 @@
+ * the source account, but not by the fee payer.
+ */
+const signedTransaction = await partiallySignTransactionMessageWithSigners(transactionMessageWithLookupTables);
+log.info(`[setup] Partially signed the transaction with a signature from ${SOURCE_ACCOUNT_SIGNER.address}`);
+
+/**
+ * SETUP: ENCODE THE TRANSACTION AS BASE64
+ * This is a common format to send a transaction between different systems
+ * If you wanted to convert it to a byte array instead, you could do this:
+ *
+ * ```ts
+ * const transactionEncoder = getTransactionEncoder();
+ * const transactionBytes = transactionEncoder.encode(signedTransaction);
+ * ```
+ */
+const base64EncodedTransaction = getBase64EncodedWireTransaction(signedTransaction);
+log.info(`[setup] Encoded the transaction as base64: ${base64EncodedTransaction.slice(0, 64)}...`);
+
+/**
+ * SETUP COMPLETE
+ * At this point, imagine we've received `base64EncodedTransaction` as input and don't know anything about it
+ * Let's use web3js to decode and inspect it
+ */
+
+/**
+ * STEP 1: DECODE TO TRANSACTION
+ * @solana/web3.js has encoders/decoders for many Solana data structures and common data formats,
+ * including both base64 strings and our `Transaction` data structure
+ * To convert between these, we first encode to a byte array, and then decode to the
+ * desired data structure.
+ * In our case we first encode the `base64EncodedTransaction` to a byte array, and then
+ * decode to a `Transaction`.
+ * If we had instead received a byte array `transactionBytes` then we would just skip
+ * the base64 encode step
+ */
+const base64Encoder = getBase64Encoder();
+const transactionBytes = base64Encoder.encode(base64EncodedTransaction);
+
+const transactionDecoder = getTransactionDecoder();
+const decodedTransaction = transactionDecoder.decode(transactionBytes);
+log.info('[step 1] Decoded the transaction');
+
+/**
+ * First let's inspect the signatures on our decoded transaction
+ * `signatures` is a map from `Address` to `SignatureBytes | null`
+ * If the address has signed the transaction then we can access their signature
+ * If not then `null` will be stored
+ */
+
+for (const [address, maybeSignature] of Object.entries(decodedTransaction.signatures)) {
+    if (maybeSignature) {
+        log.info(`[step 1] ${address} has signed the transaction`);
+    } else {
+        log.info(`[step 1] ${address} is required to sign the transaction but hasn't yet`);
+    }
+}
+
+// Now we know that we have a signature for `ED1WqT2hWJLSZtj4TtTdoovmpMrr7zpkUdbfxmcJR1Fq`
+// Let's verify that it's correct
+const signedByAddress = 'ED1WqT2hWJLSZtj4TtTdoovmpMrr7zpkUdbfxmcJR1Fq' as Address;
+const signedBySignature = decodedTransaction.signatures[signedByAddress]!;
+// We encode the source address to bytes
+const sourceAddressBytes = getAddressEncoder().encode(signedByAddress);
+// Then we create a public Ed25519 key with those bytes
```

Comment:
Wouldn't hurt to briefly plug in that this is a SubtleCrypto `CryptoKey` object, and you've designated that you want this key to have role `verify`.